### ISSUE TITLE
Date handler and field with value in format 0000-00-00 00:00:00

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#4073]
## Question
Could you advise me how to work with the date field in new admin ui?

The field is defined as string:

`    date  datetime NOT NULL DEFAULT '0000-00-00 00:00:00',`

**The option to convert it to integer in the database is not possible.**  

I would like to use new e107 date handler (calender option)
 
How to do it?

## Additional Context
My only idea, for now, is to use the type  _method_.  Convert $current value to time form_ui 
and before saving convert back to the string. 

Better solution? Maybe there is some parameter for this.  Thanks

**Comment by CaMer0n** [comment]
@Jimmi08 Yes, use `type='date'` instead of `type='datestamp'` and then in the writeParms set the type and format. 
eg. 

    array('type' => 'datetime', 'format' => 'MM, dd, yyyy hh:ii')

**Comment by Jimmi08** [comment]
Hm, no success.

```
'date'      => array (  'title' => LAN_DATESTAMP,  
              'type' => 'date',  
              'data' => 'str', 
              'width' => 'auto', 
              'readParms' =>  array(),  
              'writeParms' =>  array('type' => 'datetime', 'format' => 'MM, dd, yyyy hh:ii'), 
              'class' => 'left',  'thclass' => 'left'),
```

Calendar works.

Result: 
![image](https://user-images.githubusercontent.com/5429548/72753584-99041880-3bc5-11ea-8869-4eddd88805dd.png)

But nevermind. 
I found that it is easier to write that date than select it with the calendar.

**Comment by Moc** [comment]
What happens when you do the same for readParms as what you did with writeParms?

**Comment by Jimmi08** [comment]
Nothing, It's displayed without any change and just resaving (no change) will set it as zero.  PHP 5.6
Generated admin_config with plugin builder. No customization.

**Comment by CaMer0n** [comment]
@Jimmi08 Hmm.. Here's the code from the birthday field. Does it work for you?

```
<input class="tbox e-date input-xlarge form-control" type="text" size="40" id="e-datepicker-ue-user-birthday" value="2017-01-11" data-date-unix="false" data-date-format="yyyy-mm-dd" data-date-ampm="false" data-date-language="en" data-date-firstday="0" />
```

**Comment by Jimmi08** [comment]
@CaMer0n  It works. PHP 7.3

**Comment by CaMer0n** [comment]
@Jimmi08 Thanks. Perhaps the issue is with include the time.
